ansible_collections/arista/avd/roles/eos_designs/docs/L3 Edge.md

```diff
@@ -0,0 +1,188 @@
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;qos_profile</samp>](## "l3_edge.p2p_links.[].qos_profile") | String |  |  |  | QOS service profile. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;macsec_profile</samp>](## "l3_edge.p2p_links.[].macsec_profile") | String |  |  |  | MAC security profile. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;profile</samp>](## "l3_edge.p2p_links.[].profile") | String |  |  |  | P2P profile name. Profile defined under p2p_profiles. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;port_channel</samp>](## "l3_edge.p2p_links.[].port_channel") | Dictionary |  |  |  | Port-channel parameters. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mode</samp>](## "l3_edge.p2p_links.[].port_channel.mode") | String |  | active |  |  |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;nodes_child_interfaces</samp>](## "l3_edge.p2p_links.[].port_channel.nodes_child_interfaces") | List, items: Dictionary |  |  |  |  |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- node</samp>](## "l3_edge.p2p_links.[].port_channel.nodes_child_interfaces.[].node") | String | Required, Unique |  |  |  |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;interfaces</samp>](## "l3_edge.p2p_links.[].port_channel.nodes_child_interfaces.[].interfaces") | List, items: String |  |  |  | List of node interfaces. Ex.- [ 'node1 interface1', 'node1 interface2' ]. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- &lt;str&gt;</samp>](## "l3_edge.p2p_links.[].port_channel.nodes_child_interfaces.[].interfaces.[].&lt;str&gt;") | String |  |  |  |  |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raw_eos_cli</samp>](## "l3_edge.p2p_links.[].raw_eos_cli") | String |  |  |  | EOS CLI rendered directly on the point-to-point interface in the final EOS configuration. |
+
+=== "YAML"
+
+    ```yaml
+    l3_edge:
+      p2p_links_ip_pools:
+        - name: <str>
+          ipv4_pool: <str>
+          prefix_size: <int>
+      p2p_links_profiles:
+        - id: <int>
+          speed: <str>
+          ip_pool: <str>
+          subnet: <str>
+          ip:
+            - <str>
+          ipv6_enable: <bool>
+          nodes:
+            - <str>
+          interfaces:
+            - <str>
+          as:
+            - <str>
+          descriptions:
+            - <str>
+          include_in_underlay_protocol: <bool>
+          isis_hello_padding: <bool>
+          isis_metric: <int>
+          isis_circuit_type: <str>
+          isis_authentication_mode: <str>
+          isis_authentication_key: <str>
+          mpls_ip: <bool>
+          mpls_ldp: <bool>
+          mtu: <int>
+          bfd: <bool>
+          ptp_enable: <bool>
+          qos_profile: <str>
+          macsec_profile: <str>
+          profile: <str>
```

Comment:
@ClausHolbechArista I used `$refs: '#/$defs/l3_edge/keys/p2p_links'` for p2p_links_profiles, but I would like to remove the "profile" key as it's not applicable here.

Reply:
I believe the only option is to create a separate $def that includes the common settings.